source/simulator/core.cc

```diff
@@ -152,6 +153,7 @@ namespace aspect
     parameters (prm, mpi_communicator_),
     melt_handler (parameters.include_melt_transport ? new MeltHandler<dim> (prm) : NULL),
     newton_handler (parameters.nonlinear_solver == NonlinearSolver::Newton_Stokes ? new NewtonHandler<dim> () : NULL),
+    world_builder (WorldBuilder::create_world_builder<dim> (prm)),
```

Comment:
If you move the functions as I mentioned above, you can change this to the same structure as for the NewtonHandler or MeltHandler.